sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2487,59 +2538,117 @@ case class Encode(value: Expression, charset: Expression)
   """,
   since = "3.3.0",
   group = "string_funcs")
-// scalastyle:on line.size.limit
-case class ToBinary(
-    expr: Expression,
-    format: Option[Expression],
-    nullOnInvalidFormat: Boolean = false) extends RuntimeReplaceable
-  with ImplicitCastInputTypes {
-
-  override lazy val replacement: Expression = format.map { f =>
-    assert(f.foldable && (f.dataType == StringType || f.dataType == NullType))
-    val value = f.eval()
-    if (value == null) {
-      Literal(null, BinaryType)
-    } else {
-      value.asInstanceOf[UTF8String].toString.toLowerCase(Locale.ROOT) match {
-        case "hex" => Unhex(expr)
-        case "utf-8" => Encode(expr, Literal("UTF-8"))
-        case "base64" => UnBase64(expr)
```

Comment:
I know this PR is for fixing `to_binary`, but I'm wondering if these 3 functions have wrong behaviors as well?

As a Spark user, I'd expect `to_binary(input, 'hex')` should have the same behavior as `hex(input)`.

Reply:
We might want to consider changing behavior of `unhex` and `unbase64` as well in a separate ticket, e.g. under ansi flag? Since these were available for a long time I prefer not to expand scope of this ticket.

Reply:
I'd like to keep `ToBinary` as a `RuntimeReplaceable`. How about we add a `failOnError` bool flag to `Unhex` and `UnBase64`? It's only set to true by `ToBinary` so no behavior change for these 2 functions. It's basically moving your code changes to `Unhex` and `UnBase64`.

Reply:
Yes, I will make changes to utilize logic from `Unhex`, `Unbase64` and `Encode`. However, I am not sure we can keep it `RuntimeReplaceable` unless format is always constant (which it is not?).